src/connector/src/sink/batching_log_sink.rs

```diff
@@ -0,0 +1,134 @@
+                }
+                LogConsumerState::EpochBegun { curr_epoch } => {
+                    assert!(
+                        epoch >= curr_epoch,
+                        "new epoch {} should not be below the current epoch {}",
+                        epoch,
+                        curr_epoch
+                    );
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+                LogConsumerState::BarrierReceived { prev_epoch } => {
+                    assert!(
+                        epoch > prev_epoch,
+                        "new epoch {} should be greater than prev epoch {}",
+                        epoch,
+                        prev_epoch
+                    );
+                    sink_writer.begin_epoch(epoch).await?;
+                    LogConsumerState::EpochBegun { curr_epoch: epoch }
+                }
+            };
+            match item {
+                LogStoreReadItem::StreamChunk { chunk, chunk_id } => {
+                    match sink_writer
+                        .write_batch_and_try_finish(chunk, chunk_id)
+                        .await
+                    {
+                        Err(e) => {
+                            return Err(e);
+                        }
+                        // The file has been successfully written and is now visible to downstream consumers.
+                        // Truncate the file to remove the specified `chunk_id` and any preceding content.
+                        Ok(Some(chunk_id)) => {
+                            log_reader.truncate(TruncateOffset::Chunk {
+                                epoch: (epoch),
+                                chunk_id: (chunk_id),
+                            })?;
+                        }
+                        // The file has not been written into downstream file system.
+                        Ok(None) => {}
+                    }
+                }
+                LogStoreReadItem::Barrier { is_checkpoint: _ } => {
+                    let prev_epoch = match state {
+                        LogConsumerState::EpochBegun { curr_epoch } => curr_epoch,
+                        _ => unreachable!("epoch must have begun before handling barrier"),
+                    };
+
+                    // The current sink specifies a batching strategy, which means that sink decoupling is enabled; therefore, there is no need to forcibly write to the file when the checkpoint barrier arrives.
+                    // When the barrier arrives, call the writer's try_finish interface to check if the file write can be completed.
+                    // If it is completed, which means the file is visible in the downstream file system, thentruncate the file in the log store; otherwise, do nothing.
+                    if let Some(committed_chunk_id) = sink_writer.try_finish().await? {
+                        log_reader.truncate(TruncateOffset::Chunk {
```

Comment:
Have you tried running this PR with sink decouple off. I think We need to force commit if sink decouple is off and is_checkpoint is true. Otherwise, I suspect ckpt will be stuck.

Reply:
IMO, after this pr, sink decouple should always be on for file sink.

Reply:
After some offline discussion, let's ban `sink_decouple=false` for file sinks only: if user run `set sink_decouple=false `and then create file sink, it will return error in frontend.